src/topology/builder.rs

```diff
@@ -284,6 +311,10 @@ pub async fn build_pieces(
         detach_triggers.insert(name.clone(), trigger);
     }
 
+    // We should have all the data for the enrichment tables loaded now, so switch them over to
+    // readonly.
+    enrichment_tables.finish_load();
```

Comment:
How does this work with reloads? If I'm following correctly, we'll only build enrichment tables that are present in the config diff but then the load here swaps out the entire contents. So if you have two tables and one was not changed, would it be lost?

Reply:
Reloads are one of the compromises we will have to make. 

Since `ArcSwap` requires an atomic swap, we are not going to be able to just reload one table. So reloading will have to iterate over all the tables, if unchanged then clone the table, if changed then reload. Then with the new table list swap the tables around.

It does mean for a time during reloading memory usage for the enrichment tables will be doubled.

The alternative would be to stop processing events, swap out the data from the `ArcSwap`, reload the changed table. Swap the data back into the `ArcSwap`. Then restart the processing.